dbms/src/Storages/Transaction/ApplySnapshot.cpp

```diff
@@ -337,6 +337,9 @@ std::vector<DM::ExternalDTFileInfo> KVStore::preHandleSSTsToDTFiles(
 
     std::vector<DM::ExternalDTFileInfo> generated_ingest_ids;
     TableID physical_table_id = InvalidTableID;
+
+    auto region_id = new_region->id();
+    auto prehandle_task = prehandling_trace.registerTask(region_id);
```

Comment:
It means it will abort the previous pre handle task which is an expected behavior, or abort nothing if there is none. However every pre handling task will start by registerTask.

Meanwhile, an abort command will not be re-ordered after later pre handling tasks, even though they are in different threads. This is because the "abort" will only be called when the region no longer exists in the TiFlash store. So there will be no upcoming prehandle task of the same region.

However, I think I can add something to assert that behavior.

Reply:
Yes